src/client/components/Autocomplete/Autocomplete.tsx

```diff
@@ -52,6 +54,7 @@ const Autocomplete: React.FC<Props> = (props: Props) => {
     ) {
       // Disable saving if selected item is not changed when free text disabled
       if (readOnlyOptions && !changes.selectedItem) {
+        onSave({ value: items.find(({ value }) => value === inputValue) ? inputValue : '' })
```

Comment:
Should this return instead the matched value?

Reply:
yes, but always in the form {value: string}. using only string is accepted by Taxon but not by DataSource